src/ARC1-limitedApproval.lua

```diff
@@ -103,10 +104,12 @@ function transferFromLtd(from, to, amount, ...)
   assert(_allowance[pair], "not approved")
   assert(_allowance[pair] >= amount, "insufficient allowance")
 
-  _transfer(from, to, amount, ...)
+  re = _transfer(from, to, amount, ...)
   _allowance[pair] = _allowance[pair] - amount
 
   contract.event("transfer", system.getSender(), from, to, bignum.tostring(amount))
+
+  return re
 end
```

Comment:
when updating the above, it does not need a variable anymore and can use `return _transfer(...`